### PR TITLE
fix ctad warning in llvm

### PR DESCRIPTION
**Pull request by JordanMaples**
This fixes a couple warnings created during the LLVM build about lack of deduction guides for not_null and strict_not_null. 

LLVM builds are failing recently with this warning:
..\include\gsl/pointers(69,7): note: add a deduction guide to suppress this warning
class not_null
      ^
..\tests\notnull_tests.cpp(429,22): error: 'not_null' may not intend to support class template argument deduction [-Werror,-Wctad-maybe-unsupported]

Also:
I found while building with Clang on Windows that _MSC_VER is being defined by Clang, which causes compiler errors due to not being able to find gsl::details::get_terminate_handler. I aligned the preprocessor check to match the one defining get_terminate_handler.

### CODE REVIEW

File: include/gsl/pointers [hunk 1]
```diff
@@ -272,6 +272,16 @@ auto make_strict_not_null(T&& t) {
     return strict_not_null<std::remove_cv_t<std::remove_reference_t<T>>>{std::forward<T>(t)};
 }
 
+#if (defined(__cplusplus) && (__cplusplus >= 201703L))
```
[the remark this answers is not in the document]
Comment (JordanMaples) [reply include/gsl/pointers]: I'll change this to `#if (defined (__cpp_deduction_guides) && (__cpp_deduction_guides >= 20190L)`. Thanks!

File: include/gsl/pointers [hunk 2]
```diff
@@ -272,6 +272,16 @@ auto make_strict_not_null(T&& t) {
...
+
+// deduction guides to prevent the ctad-maybe-unsupported warning
```
[the remark this answers is not in the document]
Comment (JordanMaples) [reply include/gsl/pointers]: Yeah, I'll file that right now and try to address it next week.

File: include/gsl/pointers (hunk 1 above)
[the remark this answers is not in the document]
Comment (CaseyCarter) [reply include/gsl/pointers]: `201907L` is a bit extreme: this deduction guide doesn't need CTAD for alias templates. `201611L` - the final C++17 value - should suffice.

[the remark this answers is not in the document]
Comment (StephanTLavavej) [reply include/gsl/pointers]: Casey is absolutely right; I forgot they updated the value, and failed again to notice how new it was.

[the remark this answers is not in the document]
Comment (JordanMaples) [reply include/gsl/pointers]: That's good to know. I'll use 201611L instead. Clang currently defines __cpp_deduction_guides as 201703L so I'm glad that we can use 201611L instead..

**Comment by JordanMaples** [comment]
The one failing appveyor test appears to be due to the github api timing out. Will push another so appveyor queues again

**Pull request merged**